idea/idea-frontend-fir/idea-fir-low-level-api/src/org/jetbrains/kotlin/idea/fir/low/level/api/file/structure/FileStructureElement.kt

```diff
@@ -173,7 +174,10 @@ internal class NonReanalyzableDeclarationStructureElement(
                 firFile,
                 onDeclarationEnter = { firDeclaration ->
                     when {
-                        firDeclaration.isGeneratedDeclaration -> DiagnosticCollectorDeclarationAction.SKIP
+                        firDeclaration.isGeneratedDeclaration &&
```

Comment:
Can we just use `SKIP_CURRENT_DECLARATION_AND_CHECK_NESTED` here? I believe this will let us skip the generated declaration itself (e.g., the anonymous object) but get to the real declarations inside (e.g., the nested class in the enum entry).